### PR TITLE
Bioconda support 

### PR DESCRIPTION
**Pull request by apeltzer**
This should address https://github.com/nf-core/chipseq/issues/2 

- Provide an `environment.yaml`
- Adjust the script calls in the `main.nf` file
- Add a `run_SPP.R` script to `bin` to make this available (as it is not part of the `r-spp` package) 

The respective bioconda recipes for `r-ngsplot`, `r-ngsplotdb-(hg19,hg38,mm10)` were also added to bioconda recently and should work now :-)

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -102,7 +102,7 @@ if (params.help){
 // Configurable variables
 params.name = false
 params.project = false
-params.genome = false
+params.genomes = false
```
Comment (ewels) [inline main.nf]: Why this change? Was this intended?

File: main.nf [hunk 2]
```diff
@@ -239,8 +252,8 @@ log.info summary.collect { k,v -> "${k.padRight(21)}: $v" }.join("\n")
 log.info "===================================="
 
 // Check that Nextflow version is up to date enough
-// try / throw / catch works for NF versions < 0.25 when this was implemented
-nf_required_version = '0.28.0'
+// try / throw / catch works for NF versions < 0.30.0 when this was implemented
```
Comment (ewels) [inline main.nf]: The `nextflow.version.matches` function was still implemented in nextflow version `0.25`, that hasn't changed ;)

File: main.nf [hunk 3]
```diff
@@ -366,7 +379,7 @@ process bwa {
 
     input:
     file reads from trimmed_reads
-    file index from bwa_index.first()
+    file index from bwa_index
```
Comment (ewels) [inline main.nf]: Are you sure you don't need the `.first()`? I think this was to prevent the process from consuming the channel so that it works with more than one sample..

File: .travis.yml [hunk 4]
```diff
@@ -16,13 +19,17 @@ install:
...
   - cd /tmp/nf-core-tools
   - pip install --user -e .
-  # Reset
+  # Make test directories
+  - mkdir ${TRAVIS_BUILD_DIR}/tests
   - cd ${TRAVIS_BUILD_DIR}/tests
 
 env:
-  - NXF_VER=0.27.6
-  - ''
```
Comment (ewels) [inline .travis.yml]: Need the empty string to try testing with the latest version of Nextflow, in addition to the minimum required.

File: .travis.yml [hunk 5]
```diff
@@ -16,13 +19,17 @@ install:
...
+  - NXF_VER=0.30.1
```
Comment (ewels) [inline .travis.yml]: Do we really need v0.30.1? The pipeline should work with v0.30.0 right?

File: Singularity [hunk 6]
```diff
@@ -0,0 +1,18 @@
+From:nfcore/base
+Bootstrap:docker
+
+%labels
+    MAINTAINER Alexander Peltzer <alexander.peltzer@qbic.uni-tuebingen.de>
+    DESCRIPTION Container image containing all requirements for the nf-core/chipseq pipeline
+    VERSION 1.5dev
+
+%environment
+    PATH=/opt/conda/envs/nfcore-chipseq-1.1dev/bin:$PATH
```
Comment (ewels) [inline Singularity]: Should be `nfcore-chipseq-1.4dev`

File: main.nf (hunk 3 above)
Comment (apeltzer) [reply main.nf]: Ok, reverted that one... I had some weird errors when running the test profile and thought this caused exactly this issue, but maybe that was something different :-/

File: .travis.yml (hunk 5 above)
Comment (apeltzer) [reply .travis.yml]: Yup :-)

File: Singularity (hunk 6 above)
Comment (apeltzer) [reply Singularity]: Thanks - added that!

**Comment by apeltzer** [comment]
An issue I can't really say anything about right now: 

In the former version of th epipeline, the test script calls something like ``--genome EF4`` (which is not documented and I can't find that one nowhere. Any ideas what this is and how I could use this as test reference genome?

**Comment by ewels** [comment]
NF minimum version needs to be at least `0.30.0` because you added conda support in this PR. That was only added in v0.30.0 😉

**Comment by apeltzer** [comment]
Ah, you're right - yes I will add that back!

**Comment by ewels** [comment]
EF4 is part of iGenomes - _Saccharomyces cerevisiae_ Ensembl build.

```
s3://ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/EF4/Sequence/WholeGenomeFasta/
```

**Comment by apeltzer** [comment]
Will finish this tomorrow:

``` 
The command "nf-core lint ${TRAVIS_BUILD_DIR}" exited with 1.
3.45s$ nextflow run ${TRAVIS_BUILD_DIR} -profile test,docker --saveReference
/usr/local/bin/nextflow: line 106: cd: ../jre/bin: No such file or directory
N E X T F L O W  ~  version 0.30.0
Launching `/home/travis/build/nf-core/chipseq/main.nf` [distracted_snyder] - revision: f4de057351
ERROR ~ Unknown config attribute: igenomes_base -- check config file: /home/travis/build/nf-core/chipseq/nextflow.config
null
 -- Check '.nextflow.log' file for details
``` 

So probably just iGenomes that I need to configure properly...

**Comment by ewels** [comment]
The iGenomes error is due to a change in how nextflow parsed the params I think. We had the same thing in all pipelines anyway. Just define it in the params section in `base.config`, as done here: https://github.com/nf-core/methylseq/blob/master/conf/base.config#L102

**Comment by pditommaso** [comment]
There are some glitches on param config with 0.30.0, use 0.30.1. Also put the `params` block at the beginning of the file (ie. before `params` is referenced`).

**Comment by apeltzer** [comment]
Thanks @ewels and @pditommaso ! I adjusted this accordingly and hope that the testcases now make it through - also had a typo in my last commit yesterday :-(

**Comment by apeltzer** [comment]
So apparently the build crashes on TravisCI and locally. Most things run through but the script `run_spp.r` is not found in the step `phantompeakqualtools`. This should be automatically symlinked to the respective work dir right? Once I copy it in and run the `.command.sh` in the container, it looks good. @pditommaso some idea?

**Comment by ewels** [comment]
Not symlinked, the `bin` directory is on the `PATH` when you run the pipeline I believe..

**Comment by apeltzer** [comment]
Hmm, the error that our version number is not numeric (1.4dev) is okay I guess, but the `phantompeakqualtools` process script works fine when running it manually in a container

**Comment by ewels** [comment]
Yes, this is because you're trying to merge into `master`, so the tests expect it to be code ready for a release. As we don't have a release yet it doesn't really make a lot of sense to have a `dev` branch yet I guess.

I'll take a look at `phantompeakqualtools`..

**Comment by apeltzer** [comment]
Perfect, I guess this is the last missing piece :-)

**Comment by ewels** [comment]
I'm adding in a couple of other minor visual tweaks too whilst I'm waiting for the test run to complete ;)

I'll push direct to the bioconda branch here so my commits should append to this PR.

**Comment by apeltzer** [comment]
Perfect! I stopped doing analysis work and will do much more development in the next weeks, so eager to learn ;-)

**Comment by ewels** [comment]
Note that we could do with finding better test data really. It's yeast which is good for small reference genomes, but that means that the MACS2 process doesn't run. Seeing as we have iGenomes, may be better to generate some subsampled Human data instead so that we actually test the full pipeline...

**Comment by apeltzer** [comment]
Agreed - shouldn't be that difficult to do this. + the "blacklisting" part isn't tested on this either...

**Comment by ewels** [comment]
Ok, I fixed the `phantompeakqualtools` problem - I removed the `RScript` bit and it works fine. I guess that the R installation is using a different `PATH` or something odd. To make it work when called directly I added the env header to the script to define it as an R script.

I'm still getting an error with that process, but it is at least running now:

```
Command exit status:
  1

Command output:
  ################
  ChIP data: SRR5204809_Spt5-ChIP_Input1_SacCer_ChIP-Seq_ss100k.dedup.sorted.bam
  Control data: NA
  strandshift(min): -500
  strandshift(step): 5
  strandshift(max) 1500
  user-defined peak shift NA
  exclusion(min): 10
  exclusion(max): NaN
  num parallel nodes: NA
  FDR threshold: 0.01
  NumPeaks Threshold: NA
  Output Directory: .
  narrowPeak output file name: NA
  regionPeak output file name: NA
  Rdata filename: NA
  plot pdf filename: ./SRR5204809_Spt5-ChIP_Input1_SacCer_ChIP-Seq_ss100k.dedup.sorted.pdf
  result filename: SRR5204809_Spt5-ChIP_Input1_SacCer_ChIP-Seq_ss100k.spp.out
  Overwrite files?: FALSE

  [1] TRUE
  Reading ChIP tagAlign/BAM file SRR5204809_Spt5-ChIP_Input1_SacCer_ChIP-Seq_ss100k.dedup.sorted.bam

Command error:
  Loading required package: Rcpp
  awk: line 2: function and never defined
  Error converting BAM to tagalign file: /tmp/RtmpD4OZmr/SRR5204809_Spt5-ChIP_Input1_SacCer_ChIP-Seq_ss100k.dedup.sorted.bam1b1cc6fbab
```

I also removed a lot of the ignore statements (I feel like this pipeline should be stable enough that they're not needed). And updated the logo in a couple of places.

**Comment by ewels** [comment]
..and it seems that my editor cleaned up a bunch of whitespace, sorry about that.

I also noticed that there was a mixture of `params.version` and just `version` used in the pipeline (with different version numbers!), so I fixed that up.

**Comment by ewels** [comment]
So yeah - I think the error is due to different versions of `awk`. So hopefully something we can fix in the container easily...? 😟

**Comment by MaxUlysse** [comment]
> ..and it seems that my editor cleaned up a bunch of whitespace, sorry about that.
I think that's good, you should not excuse yourself about that ;-)

**Comment by ewels** [comment]
Ok, phantompeakqualtools now works for me! I see now that `gawk` was already added in this PR, so I guess you'd already come across this and fixed it yourself @apeltzer. But as my container was out of date I didn't have it.

I did a couple of other minor fixes such as forcing the run_spp.r script to import the R libraries it needs, and now the whole pipeline seems to be running through for me 🎉  (still waiting for the final steps).

Note that the CI tests will fail because the docker image they're running with isn't the latest one from this PR.
